src/main/java/cinemas/services/impl/CityServiceImpl.java

```diff
@@ -0,0 +1,47 @@
+package cinemas.services.impl;
+
+import cinemas.dtos.CityDto;
+import cinemas.dtos.TheaterDto;
+import cinemas.models.City;
+import cinemas.repositories.CityRepository;
+import cinemas.services.CityService;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class CityServiceImpl implements CityService {
+    private CityRepository cityRepository;
+
+    public CityServiceImpl(CityRepository cityRepository) {
+        this.cityRepository = cityRepository;
+    }
+
+    @Override
+    public List<CityDto> getAllCityDtos() {
+        var cities = cityRepository.findAll();
+        var cityDtos = new ArrayList<CityDto>();
+        cities.forEach(city -> {
+            cityDtos.add(new CityDto(city.getId(), city.getName(), null));
+        });
+        return cityDtos;
+    }
+
+    @Override
+    public City getCityById(int id) {
+        return cityRepository.findById(id).orElse(null);
+    }
+
+    @Override
+    public CityDto getCityDtoWithTheatersById(int id) {
+        var city = cityRepository.findById(id);
+        if (city.isEmpty()) {
```

Comment:
ủa sao lại là isEmpty nhỉ
ở đây city là 1 obejct có value hoặc là null mà

chị cũng thấy findById sẽ trả về 1 object hoặc null cơ mà em nhỉ

Reply:
![image](https://github.com/user-attachments/assets/3a24cade-da12-4b8a-91ff-1e073c9a3a99)
dạ theo định nghĩa thì nó sẽ trả về object vs value nếu not-null, còn null thì là optional vs empty value ạ